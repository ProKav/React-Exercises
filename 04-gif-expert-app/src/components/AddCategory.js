import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('')

	const handleInputChange = e => {
		setInputValue(e.target.value)
	}

	const handleSubmit = e => {
		e.preventDefault()

		if (inputValue.trim().length > 2) {
			setCategories(categ => [inputValue, ...categ])
			setInputValue('')
		} else {
			setInputValue('')
		}
	}

	return (
		<form className='form__group field' onSubmit={handleSubmit}>
			<br />
			<p>{inputValue}</p>
			<input
				className='form__field'
				name='name'
				id='name'
				type='text'
				value={inputValue}
				onChange={handleInputChange}
			/>
			<label htmlFor='name' className='form__label'>
				Type Here...
			</label>
		</form>
	)
}

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
}
