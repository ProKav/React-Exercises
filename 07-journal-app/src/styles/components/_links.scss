.link {
	color: $dark-grey;
	text-decoration: none;
	padding: 10px;

	&:hover {
		text-decoration: underline;
	}
}
