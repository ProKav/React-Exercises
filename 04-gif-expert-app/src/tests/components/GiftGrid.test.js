import React from 'react';
import { GiftGrid } from '../../components/GiftGrid';
import { shallow } from 'enzyme';
import { useFetchGifts } from '../../hooks/useFetchGifts';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifts');

describe('Tests for GiftGrid component', () => {
	const category = 'Goku';
	// const wrapper = shallow(<GifGrid category={category}/>);

	test('should show the correct component and match with the snapshot', () => {
		useFetchGifts.mockReturnValue({
			data: [],
			loading: true,
		});
		const wrapper = shallow(<GiftGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should change the snapshot when useFetchGifs is executed', () => {
		const gifs = [
			{
				id: 'ABC',
				url: 'https://google.com',
				title: 'ABC Title',
			},
			{
				id: '123',
				url: 'https://google.com',
				title: 'ABC Title',
			},
		];

		useFetchGifts.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GiftGrid category={category} />);
		// expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GiftGridItem').length).toBe(gifs.length);
	});
});
