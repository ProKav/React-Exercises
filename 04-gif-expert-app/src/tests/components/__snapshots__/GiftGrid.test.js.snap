// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GiftGrid component should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className=" animate__animated animate__flash"
  >
    Loading
  </p>
  <br />
  <div
    className="card-grid"
  />
</Fragment>
`;
